scenarios/aws/microVMs/microvms/libvirt.go

```diff
@@ -177,39 +178,59 @@ func (vm *VMCollection) SetupCollectionDomainConfigurations(depends []pulumi.Res
 			}
 		}
 
-		vm.domains = append(vm.domains, domains...)
+		if _, ok := vm.domains[set.ID]; ok {
+			vm.domains[set.ID] = append(vm.domains[set.ID], domains...)
+		} else {
+			vm.domains[set.ID] = domains
+		}
 	}
 
 	return waitFor, nil
 }
 
 func (vm *VMCollection) SetupCollectionNetwork(depends []pulumi.Resource) error {
-	var dhcpEntries []interface{}
-	var err error
-
-	for _, d := range vm.domains {
-		dhcpEntries = append(dhcpEntries, d.dhcpEntry)
-
+	dhcpEntries := make(map[vmconfig.VMSetID][]interface{})
+	for setID, dls := range vm.domains {
+		for _, domain := range dls {
+			if _, ok := dhcpEntries[setID]; ok {
+				dhcpEntries[setID] = append(dhcpEntries[setID], domain.dhcpEntry)
+			} else {
+				dhcpEntries[setID] = []interface{}{domain.dhcpEntry}
+			}
```

Comment:
Same as the other one:

```suggestion
			dhcpEntries[setID] = append(dhcpEntries[setID], domain.dhcpEntry)
```